planex/extract.py

```diff
@@ -98,8 +98,11 @@ def archive_root(tar):
     """
     Return the name of the top level directory of the tarball
     """
-    if tar.firstmember.type == tarfile.DIRTYPE:
-        return tar.firstmember.name
+    topname = os.path.commonprefix(tar.getnames())
```

Comment:
This all looks good. My only comment would be that you are calling `tar.getnames() twice. If this is an expensive operation it would make sense to store the result in a variable.